### PR TITLE
add http example (zip-codes)

### PR DESCRIPTION
**Pull request by ericgj**
This is a translation of [Elm's "zip codes" example](http://elm-lang.org/examples/zip-codes), designed to show one way of handling asychronous tasks -- specifically, http requests.  

It's not my favorite example for several reasons -- for one thing, it's very US-centric -- for another, being 'stateless' it's not very representative of most apps. But I think it could be useful for people to see side-by-side with the original. A lot of the ceremony needed by Elm (ports, mailboxes, etc., which I don't really understand very well yet) I found could be simplified since we're directly in the browser context.

But would appreciate your views on it and on dealing with async tasks in general using this architecture, & of course feel free to modify this example as you see fit.

### CODE REVIEW

File: examples/zip-codes/js/app.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    },
+    []
+  )
+ 
+  const messages = Result.case({
+    Ok:  R.map((city) => h('div', {style: myStyle}, city)),
+    Err: (msg) => [ h('div', { style: myStyle }, msg) ]
+  }, result);
+  
+  return h('div', {}, R.concat([field], messages));
```
Comment (paldepind) [inline examples/zip-codes/js/app.js]: What about using [prepend](http://ramdajs.com/0.15/docs/#prepend) here?

File: examples/zip-codes/js/app.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+
+function lookupZipCode(query){
+  const toRequest = (success,fail) => {
+    if (query.match(/^\d{5}$/)) { 
+      success( new Request("http://api.zippopotam.us/us/" + query, {method: 'GET'}) ); 
+    } else { 
+      fail( "Give me a valid US zip code!" ); 
+    }
```
Comment (paldepind) [inline examples/zip-codes/js/app.js:82]: This is [The Deferred anti-pattern](https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern). You're creating a promise at line 84 just to create another promise inside it. You can get rid of the `new Promise` by using either the promise from fetch or one from `Promise.reject`.

Comment (ericgj) [inline examples/zip-codes/js/app.js:82]: Yeah. I just added a commit reworking this. I don't quite get all the eccentricities of JS promises/fetch, haven't worked with them much. But just to note that the `.then(promise)` I still use in rejecting 404 errors -- it seems to be standard practice (!) at least according to [this tutorial](http://updates.html5rocks.com/2015/03/introduction-to-fetch).

**Comment by paldepind** [comment]
Looks splendid! Thank you for making this.

**Comment by paldepind** [comment]
Oops. I pressed the wrong button ;)

**Pull request merged**